src/cursor.rs

```diff
@@ -27,11 +28,18 @@ struct RawCursorLoading {
 }
 
 impl RawCursor {
-    fn new(response: Response) -> Self {
+    pub(crate) fn new(response: Response) -> Self {
         Self(RawCursorInner::Waiting(response.into_future()))
     }
 
-    async fn next(&mut self) -> Result<Option<Bytes>> {
+    pub fn newline(self) -> RawCursorNewline {
```

Comment:
The `futures` crate already provides  the [https://docs.rs/futures/latest/futures/io/trait.AsyncBufReadExt.html#method.lines](AsyncBufReadExt::lines) method.

It seems we can implement the `AsyncBufRead` (under the `futures03` feature) trait for `RawCursor` covering more use cases.